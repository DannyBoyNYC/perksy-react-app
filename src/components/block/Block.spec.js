const React = require('react');

const Block = require('./index').default;

describe('<Block.* />', () => {
  let component;
  const requiredProps = {
    items: [
      {
        name: 'Foo',
        respondents: {
          current: 100
        }
      },
      {
        name: 'Bar',
        respondents: {
          current: 1000
        }
      },
      {
        name: 'Baz',
        respondents: {
          current: 10000
        }
      }
    ]
  };

  describe('<Block.CampaignPercentage />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Block.CampaignPercentage {...requiredProps} />,
        'Block'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render a Campaign.Percentage for each item', () => {
      expect(component).toContainMatchingElements(3, 'Percentage');
    });

    it('should render a heading if given one', () => {
      component.setProps({
        heading: 'Title content'
      });
      expect(component).toIncludeText('Title content');
    });

    it('should render a div with the word `None` if items is empty', () => {
      component.setProps({
        items: []
      });
      expect(component).toHaveText('None');
    });
  });

  describe('<Block.CampaignRow />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Block.CampaignRow {...requiredProps} />,
        'Block'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render a Campaign.Row for each item', () => {
      expect(component).toContainMatchingElements(3, 'Row');
    });
  });
});
