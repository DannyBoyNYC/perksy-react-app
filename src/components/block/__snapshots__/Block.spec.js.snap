// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block.* /> <Block.CampaignPercentage /> should render correctly with all required props 1`] = `
<div
  className="Block-root-0-1-1 Block-campaign-percentage-0-1-5"
>
  <Percentage
    className="Block-item-0-1-4"
    key="block-0"
    name="Foo"
    respondents={
      Object {
        "current": 100,
      }
    }
  />
  <Percentage
    className="Block-item-0-1-4"
    key="block-1"
    name="Bar"
    respondents={
      Object {
        "current": 1000,
      }
    }
  />
  <Percentage
    className="Block-item-0-1-4"
    key="block-2"
    name="Baz"
    respondents={
      Object {
        "current": 10000,
      }
    }
  />
</div>
`;

exports[`<Block.* /> <Block.CampaignRow /> should render correctly with all required props 1`] = `
<div
  className="Block-root-0-1-1 Block-campaign-row-0-1-6"
>
  <Row
    className="Block-item-0-1-4"
    key="block-0"
    name="Foo"
    respondents={
      Object {
        "current": 100,
      }
    }
  />
  <Row
    className="Block-item-0-1-4"
    key="block-1"
    name="Bar"
    respondents={
      Object {
        "current": 1000,
      }
    }
  />
  <Row
    className="Block-item-0-1-4"
    key="block-2"
    name="Baz"
    respondents={
      Object {
        "current": 10000,
      }
    }
  />
</div>
`;
